qiskit/transpiler/passes/optimization/inverse_cancellation.py

```diff
@@ -58,12 +58,16 @@ def __init__(self, gates_to_cancel: List[Union[Gate, Tuple[Gate, Gate]]]):
 
         self.self_inverse_gates = []
         self.inverse_gate_pairs = []
+        self.self_inverse_gate_names = set()
+        self.inverse_gate_pairs_names = set()
 
```

Comment:
Do we actually need the gate instances?  As far as I could see, we only ever inspect the names.

Reply:
Oh that's a good point, I just assumed we needed the instances and just stored the names separately. But you're right nothing ever uses the gate instances outside of `__init__` to check things are actually inverses of each other.

Reply:
I looked into this and we do need the gate instances so that we can handle parameterized gates. The inverse pairs function uses this now, the self inverse path doesn't but this is a bug and I'm going to fix it.